tools/azure-sdk-tools/packaging_tools/code_report.py

```diff
@@ -221,6 +241,12 @@ def main(input_parameter: str, version: Optional[str] = None, no_venv: bool = Fa
         with open(output_filename, "w") as fd:
             json.dump(merged_report, fd, indent=2)
             _LOGGER.info(f"Merged report written to {output_filename}")
+        metadata["reports"].append(str(output_filename))
+
+    if metadata_path:
+        with open(metadata_path, "w") as metadata_fd:
```

Comment:
If you want to write something in SDK Automation, you can use the TMPDIR in environment variable. It's a dedicated tmp folder for your scripts.

Reply:
Thanks Phoenix! This code is way more generic than the automation, and I'm using a Python tmpdir then. I actually externalize this part unto a different PR:
https://github.com/Azure/azure-sdk-for-python/pull/15652